CONTRIBUTING.md

```diff
@@ -63,6 +64,21 @@ quickly merge or address your contributions.
 
 ## Pull requests
 
+Before working on a pull request, open an issue for your intended work to discuss 
```

Comment:
Sorry, I asked to make the language stronger about opening an issue and now I feel it might be too strong.  Like if it's just fixing a typo or a simple bugfix for an already open issue, maybe they don't need to open another issue?  How about:

```
When submitting a PR you should reference an existing issue.  If no issue already exists, please create one.  This can be skipped for trivial PRs like fixing typos.

Creating an issue in advance of working on the PR can help to avoid duplication of effort, e.g. maybe we know of existing related work.   Or it may be that we can provide guidance that will help with your approach.
```

Reply:
Sounds good